LEGO1/mxbitmap.cpp

```diff
@@ -0,0 +1,105 @@
+#include "mxbitmap.h"
+#include "mxbool.h"
+#include "mxpalette.h"
+
+// OFFSET: LEGO1 0x100bc980
+MxBitmap::MxBitmap()
+{
+    this->m_info = NULL;
+    this->m_bmiHeader = NULL;
+    this->m_paletteData = NULL;
+    this->m_data = NULL;
+    this->m_unk18 = 0;
+    this->m_palette = NULL;
+}
+
+// OFFSET: LEGO1 0x100bca10
+MxBitmap::~MxBitmap()
+{
+    if (this->m_info != NULL) {
+        free(this->m_info);
+    }
+    if (this->m_data != NULL) {
+        free(this->m_data);
+    }
+    if (this->m_palette != NULL) {
+        // FIXME: Once m_palette/base is understood
+    }
+    // FIXME: Unwind@100bca91. Stack unwinding, I've never done this before.
+}
+
+// OFFSET: LEGO1 0x100bd1c0
+MxPalette *MxBitmap::CreatePalette()
+{
+    // FIXME: This function needs MxPalette to be completed. Also INFERRING usage of MxBool
+    MxPalette *pal;
+    MxPalette *ppal;
+    MxBool success;
+
+    pal = NULL;
+    success = MX_FALSE;
+    if(this->m_unk18 == '\0') {
+        ppal = (MxPalette *)malloc(0x414);
+        pal = NULL;
+        if(ppal != NULL) {
+            // pal = MxPalette::FromBitmapPalette(ppal, this->m_paletteData);
+        }
+    } else {
+        if(this->m_unk18 != '\x01') {
+            goto LAB_100bd253;
+        }
```

Comment:
formatting + change the code in a way to get rid of the `goto`'s.
The goto's were added by the Ghidra decompiler, but weren't there in the OG code.

Reply:
I see the goto call; is it a call to another function?